tests/templates/kuttl/kerberos-ad/01-install-secretclass.yaml

```diff
@@ -2,7 +2,7 @@
 apiVersion: kuttl.dev/v1beta1
 kind: TestStep
 commands:
-  - script: yq eval '.metadata.name = "kerberos-" + strenv(NAMESPACE) | .spec.backend.kerberosKeytab |= (.adminKeytabSecret.namespace = strenv(NAMESPACE) | .admin.activeDirectory.passwordCacheSecret.namespace = .adminKeytabSecret.namespace | .admin.activeDirectory.ldapTlsCaSecret.namespace = .adminKeytabSecret.namespace)' secretclass.yaml | kubectl apply -f-
+  - script: eval "echo \"$(cat secretclass.yaml)\"" | kubectl apply -f -
```

Comment:
If I understand this right, the shell substitution would happen in the outer shell, right? Wouldn't that mean that this breaks as soon as the YAML contains a double quote?

Reply:
> If I understand this right, the shell substitution would happen in the outer shell, right?

Yes, in the shell started by KUTTL.

> Wouldn't that mean that this breaks as soon as the YAML contains a double quote?

Double quotes are removed. If they are required then they must be escaped:

```
$ cat test.yaml
unquotedNamespace: $NAMESPACE.suffix
quotedNamespace: "$NAMESPACE.suffix"
requiredQuotes: \"1\"

$ NAMESPACE=mynamespace sh -c 'eval "echo \"$(cat test.yaml)\""'
unquotedNamespace: mynamespace.suffix
quotedNamespace: mynamespace.suffix
requiredQuotes: "1"
```